app/calendar/edit-habit.tsx

```diff
@@ -0,0 +1,97 @@
+import { useSWRConfig, Fetcher } from "swr";
+import { useState } from "react";
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog";
+import { Label } from "@/components/ui/label";
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+import { useSession } from "next-auth/react";
+import { Dispatch, SetStateAction } from "react";
+
+export default function EditHabit({
```

Comment:
Let's use `HabitEditor` as the name of the new component. The reason for this is to be consistent with other components